app/controllers/concerns/foreman/controller/puppet/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,164 @@
+module Foreman::Controller::Puppet::HostsControllerExtensions
+  extend ActiveSupport::Concern
```

Comment:
A new controller is a way to introduce a new resource to the system (a whole MVC stack) - right now, and even with the introduction of facets, we will be dealing with host as a main resource. We don't want to make facets as a standalone first-rate citizens, just a dependent objects of the whole host.
The UI as it's build today does not allow this separation either, since host screens a de-facto aggregated from different facets each host has.
I do agree that actions that are not UI-dependent (and I can think only about `#puppetrun`) could be extracted to a different controller. I think it's too small amount of code to move right now.
If we ever going to change the UI of our host to be more block oriented, so each block could be provided by different facet - that could be a good place for creating a separate REST resource for facets. IMHO it's a matter for another PR (maybe even RFC).